### PR TITLE
Update deprecated protocol feature components (Issue #10341)

### PR DESCRIPTION
**Pull request by maureenlholland**
## Description

`firefox/developer` pages use Protocol's Feature Card, which will be deprecated. We want to replace this component with Split where possible. There is also a deprecated Picto Card that has been replaced with Picto.

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/10341

## Testing

New split components to compare with existing feature components:
light-bg
dark-bg

New picto component to compare with existing picto card component:
picto-card

http://localhost:8000/en-US/firefox/developer/
http://localhost:8000/en-US/firefox/91.0a2/whatsnew/all/
http://localhost:8000/en-US/firefox/91.0a2/firstrun
- (WNP & firstrun only) click Next Gen link in top section, expect to arrive at light bg split for "The browser made for developers"

Conditional statements to switch between `for-developers` and `performance`:
- in `firefox/developer/includes/whatsnew.html` at L43
- in `firefox/developer/index.html` at L69

### CODE REVIEW

File: bedrock/firefox/templates/firefox/developer/includes/performance.html [hunk 1]
```diff
@@ -1,21 +1,12 @@
-<section class="mzp-l-content t-performance" id="nextgen">
```
Comment (maureenlholland) [inline bedrock/firefox/templates/firefox/developer/includes/performance.html:1]: note: may need to restore id

Comment (maureenlholland) [reply bedrock/firefox/templates/firefox/developer/includes/performance.html:1]: ID restored

**Review by craigcook — APPROVED** [review]
This looks great, excellent work! A few minor redundancies we could tidy up if you want to, but they're not significant. And it's up to you if you want to add an ID to the Split macro as part of this or in a separate PR.

So an initial r+ and I can review any updates if you make them, but also happy to merge this as-is.

File: bedrock/firefox/templates/firefox/developer/includes/performance.html [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
-          </p>
-          <p class="mzp-c-cta-link"><a href="https://hacks.mozilla.org/2017/08/inside-a-super-fast-css-engine-quantum-css-aka-stylo/">{{ ftl('ui-learn-more') }}</a></p>
-        </div>
-      </div>
-    </div>
-  </div>
-</section>
+{% from "macros-protocol.html" import split with context %}
+
+<div id="nextgen">
```
[the remark this answers is not in the document]
Comment (maureenlholland) [reply bedrock/firefox/templates/firefox/developer/includes/performance.html]: Added! I also saw a `background_class` conditional but think it was meant to be `theme_class`, so made the switch. For some reason, the closing div seemed to be added in any case for split components with theme classes (see https://www.mozilla.org/en-US/about/policy/lean-data/ & http://localhost:8000/en-US/about/policy/lean-data/)

File: bedrock/base/templates/macros-protocol.html [hunk 3]
```diff
@@ -66,8 +66,9 @@
 
 
 {# Split: https://protocol.mozilla.org/patterns/organisms/split.html #}
-{% macro split(block_class=None, theme_class=None, body_class=None, image_url=None, media_class=None, include_highres_image=False, l10n_image=False, image_alt='') -%}
-<section class="mzp-c-split{% if block_class %} {{ block_class }}{% endif %}">
+{% macro split(id=None, block_class=None, theme_class=None, body_class=None, image_url=None, media_class=None, include_highres_image=False, l10n_image=False, image_alt='') -%}
+
+<section {% if id %} id={{ id }} {% endif %} class="mzp-c-split{% if block_class %} {{ block_class }}{% endif %}">
```
Comment (craigcook) [inline bedrock/base/templates/macros-protocol.html:71]: This should include quotes around the value:
```suggestion
<section {% if id %}id="{{ id }}"{% endif %} class="mzp-c-split{% if block_class %} {{ block_class }}{% endif %}">
```

File: bedrock/base/templates/macros-protocol.html [hunk 4]
```diff
@@ -89,7 +90,7 @@
     </div>
     {% endif %}
   </div>
-  {% if background_class %}
+  {% if theme_class %}
```
Comment (craigcook) [inline bedrock/base/templates/macros-protocol.html:93]: Oops! Good catch.

**Review by craigcook — CHANGES_REQUESTED** [review]
One teeny nitpick to add quotes, then I think we're good to go.

**Review by craigcook — APPROVED** [review]

**Pull request merged**